File: Assessment/Assessment.WEB/src/app/services/student.service.ts
import { Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Student } from '../entities/student';

export class StudentService {
    http: Http;
    baseUrl: string;
    controller: string = "api/Student/";

    constructor(http: Http, baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public getAll() {
        return this.http.get(this.baseUrl + this.controller);
    }

    public get(id: number) {
        return this.http.get(this.baseUrl + this.controller + id);
    }

    public update(student: Student) {
        return this.http.put(this.baseUrl + this.controller, student);
    }

    public insert(student: Student) {
        return this.http.post(this.baseUrl + this.controller, student);
    }

    public delete(student: Student) {
        return this.http.delete(this.baseUrl + this.controller);
    }
}

File: Assessment/Assessment.Entities/StudentService.cs
using System;
using System.Collections.Generic;
using Assessment.Entities.DataInterface;

namespace Assessment.Entities
{
    public class StudentService
    {
        private IStudentRepository _repository;

        public StudentService(IStudentRepository repository)
        {
            _repository = repository;
        }

        public void Insert(Student student)
        {
            _repository.Insert(student);
        }

        public void Delete(Student student)
        {
            _repository.Delete(student);
        }

        public void Update(Student student)
        {
            _repository.Update(student);
        }

        public Student Get(int studentID)
        {
            return _repository.Get(studentID);
        }

        public IEnumerable<Student> GetAll()
        {
            return _repository.GetAll();
        }
    }
}